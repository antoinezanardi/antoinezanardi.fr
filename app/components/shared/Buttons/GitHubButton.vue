<template>
  <a
    :aria-label="$t('GitHubButton.discoverMyProjects')"
    class="btn btn-icon btn-lg btn-primary btn-round"
    data-bs-toggle="tooltip"
    href="https://github.com/antoinezanardi"
    rel="noopener noreferrer"
    target="_blank"
    :title="$t('GitHubButton.discoverMyProjects')"
  >
    <WrappedFontAwesomeIcon
      icon="fa-brands fa-github"
      icon-color="#000000"
      size="2x"
    />
  </a>
</template>

<script setup lang="ts">
import WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";
</script>