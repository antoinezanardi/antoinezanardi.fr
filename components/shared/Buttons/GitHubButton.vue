<template>
  <a
    class="btn btn-primary btn-round btn-lg btn-icon"
    href="https://github.com/antoinezanardi"
    target="_blank"
    rel="noopener noreferrer"
    data-bs-toggle="tooltip"
    :title="$t('GitHubButton.discoverMyProjects')"
  >
    <i class="fa-brands fa-github fa-2x fa-btn"/>
  </a>
</template>