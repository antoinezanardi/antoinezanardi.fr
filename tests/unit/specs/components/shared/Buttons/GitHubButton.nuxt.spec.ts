import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

import GitHubButton from "@/components/shared/Buttons/GitHubButton.vue";

describe("GitHub Button Component", () => {
  let wrapper: ReturnType<typeof mount<typeof GitHubButton>>;

  async function mountGitHubButtonComponent(options: ComponentMountingOptions<typeof GitHubButton> = {}): Promise<ReturnType<typeof mount<typeof GitHubButton>>> {
    return mountSuspendedComponent(GitHubButton, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountGitHubButtonComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});