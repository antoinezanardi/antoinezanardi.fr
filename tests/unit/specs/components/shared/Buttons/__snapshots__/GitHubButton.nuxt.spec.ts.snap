// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GitHub Button Component > should match snapshot when rendered. 1`] = `
"<a data-v-badedb06="" aria-label="Découvrez mes projets" class="btn btn-icon btn-lg btn-primary btn-round" data-bs-toggle="tooltip" href="https://github.com/antoinezanardi" rel="noopener noreferrer" target="_blank" title="Découvrez mes projets">
  <wrapped-font-awesome-icon-stub data-v-badedb06="" icon="fa-brands fa-github" iconcolor="white" size="2x" classes=""></wrapped-font-awesome-icon-stub>
</a>"
`;
